test/test-with-server.js

```diff
@@ -1348,6 +1348,108 @@ describe("follow-redirects", function () {
         });
     });
   });
+
+  describe("transform request options", function () {
+    it("don't call transform function", function () {
+      app.get("/a", sendsJson({ a: "b" }));
+      var callsToTransform = 0;
+
+      return server.start(app)
+        .then(asPromise(function (resolve, reject) {
+          var options = {
+            host: "localhost",
+            port: 3600,
+            path: "/a",
+            method: "GET",
+            transformOptions: function () {
+              callsToTransform++;
+            },
+          };
+          http.get(options, concatJson(resolve, reject)).on("error", reject);
+        }))
+        .then(function (res) {
+          assert.strictEqual(callsToTransform, 0);
+          assert.deepEqual(res.parsedJson, { a: "b" });
+          assert.deepEqual(res.responseUrl, "http://localhost:3600/a");
+        });
+    });
+    it("recover, if transform function is null", function () {
```

Comment:
"ignore beforeRedirect if not a function"

and only one test is needed, just set `beforeRedirect` to `3` or so.

Reply:
done